awscli/customizations/s3/results.py

```diff
@@ -400,23 +404,33 @@ def _add_progress_if_needed(self):
 
     def _print_progress(self, **kwargs):
         # Get all of the statistics in the correct form.
-        bytes_completed = human_readable_size(
-            self._result_recorder.bytes_transferred +
-            self._result_recorder.bytes_failed_to_transfer
-        )
-        expected_bytes_completed = human_readable_size(
-            self._result_recorder.expected_bytes_transferred)
         remaining_files = str(
             self._result_recorder.expected_files_transferred -
             self._result_recorder.files_transferred
         )
 
         # Create the display statement.
-        progress_statement = self.PROGRESS_FORMAT.format(
-            bytes_completed=bytes_completed,
-            expected_bytes_completed=expected_bytes_completed,
-            remaining_files=remaining_files
-        )
+        if self._result_recorder.expected_bytes_transferred > 0:
+            bytes_completed = human_readable_size(
+                self._result_recorder.bytes_transferred +
+                self._result_recorder.bytes_failed_to_transfer
+            )
+            expected_bytes_completed = human_readable_size(
+                self._result_recorder.expected_bytes_transferred)
+
+            progress_statement = self.BYTE_PROGRESS_FORMAT.format(
+                bytes_completed=bytes_completed,
+                expected_bytes_completed=expected_bytes_completed,
+                remaining_files=remaining_files
+            )
+        else:
+            files_completed = self._result_recorder.files_transferred
```

Comment:
You do not need to take into account ``files_failed`` and ``files_warned``. ``files_transferred`` takes into account successful transfers and failed transfers. We do not want to be including warned files because those correlate directly to number of warnings seen (we probably should rename it but not for this PR) and you want to avoid double counting as a file could have produced a warning but also been transferred.